### PR TITLE
Feature/simplify single field dispatcher

### PR DESCRIPTION
**Pull request by Nasdan**

### CODE REVIEW

File: src/validation-dispatcher/single-field-dispatcher.ts [hunk 1]
```diff
@@ -3,128 +3,58 @@ import {
...
+): Promise<ValidationResult> => {
+  const validationFn = getValidationFn(fieldValidation);
+  return validationFn(value, values, fieldValidation['customArgs']).then(
+    checkValidationResult
+  );
 };
 
-export const fireSingleFieldValidations = (
-  values: any,
+const findFirstFailedValidation = (
```
Comment (brauliodiez) [inline src/validation-dispatcher/single-field-dispatcher.ts]: fireSingleFieldValidations

File: src/validation-dispatcher/single-field-dispatcher.ts [hunk 2]
```diff
@@ -3,128 +3,58 @@ import {
...
-  const fieldValidationResultPromise = new Promise<ValidationResult>(
-    (resolve, reject) => {
-      if (arrayContainsEntries(fieldValidations)) {
-        iterateFiringFieldValidations(resolve, reject, validationParams, 0);
-      } else {
-        resolve(createDefaultValidationResult());
-      }
-    }
+) =>
+  fieldValidations.reduce(
```
Comment (brauliodiez) [inline src/validation-dispatcher/single-field-dispatcher.ts]: Add this link to explain the chaining behavior: https://css-tricks.com/why-using-reduce-to-sequentially-resolve-promises-works/

File: src/validation-dispatcher/single-field-dispatcher.ts [hunk 3]
```diff
@@ -3,128 +3,58 @@ import {
...
+    (result, next) =>
+      result.then((validationResult: ValidationResult) =>
+        validationResult.succeeded
+          ? fireValidation(value, values, next)
+          : validationResult
+      ),
+    fireValidation(value, values, fieldValidations[0])
```
Comment (brauliodiez) [inline src/validation-dispatcher/single-field-dispatcher.ts]: Add four more tests including different set timeout resolutions.

File: src/validation-dispatcher/single-field-dispatcher.ts (hunk 1 above)
Comment (Nasdan) [reply src/validation-dispatcher/single-field-dispatcher.ts]: Done

File: src/validation-dispatcher/single-field-dispatcher.ts (hunk 2 above)
Comment (Nasdan) [reply src/validation-dispatcher/single-field-dispatcher.ts]: Done

File: src/validation-dispatcher/single-field-dispatcher.ts (hunk 3 above)
Comment (Nasdan) [reply src/validation-dispatcher/single-field-dispatcher.ts]: Done

**Comment by brauliodiez** [comment]
Really good stuff, kudos !

**Pull request merged**